google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/RpcMethodMappings.java

```diff
@@ -0,0 +1,1942 @@
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.bucket_acl;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.buckets;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.default_object_acl;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.hmacKey;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.object_acl;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.objects;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.serviceaccount;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.BucketAcl;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.Buckets;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.DefaultObjectAcl;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.HmacKey;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.Notification;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.ObjectAcl;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.Objects;
+import com.google.cloud.storage.conformance.retry.RpcMethodMappings.Mappings.ServiceAccount;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.MultimapBuilder;
+import com.google.common.collect.Multimaps;
+import com.google.common.collect.Sets;
+import com.google.common.io.ByteStreams;
+import com.google.errorprone.annotations.Immutable;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map.Entry;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * A class which serves to try and organize all of the {@link RpcMethodMapping} for the retry
+ * conformance tests.
+ *
+ * <p>Individual mappings are grouped via inner classes corresponding to the {@link RpcMethod} for
+ * which they are defined.
+ *
+ * <p>As part of construction mappingIds are enforced to be unique, throwing an error if not.
+ */
+@Immutable
+@SuppressWarnings("Guava")
+final class RpcMethodMappings {
+  private static final Logger LOGGER = Logger.getLogger(RpcMethodMappings.class.getName());
+
+  static final int _2MiB = 2 * 1024 * 1024;
+  final Multimap<RpcMethod, RpcMethodMapping> funcMap;
+
+  RpcMethodMappings() {
+    ArrayList<RpcMethodMapping> a = new ArrayList<>();
+
+    BucketAcl.delete(a);
```

Comment:
Mapping seems complex because it requires a separate Model class to be introduced. Could you talk more to this?

Reply:
This was addressed offline.